### PR TITLE
fix issue #484 (Remove image when no data in nutrient levels)

### PR DESCRIPTION
**Pull request by sowmya9029**
## Description

Changed NutritionProductFragment.java and fragment_nutrition_product.xml files inorder to align ui and show no data beside nutrition level.
device-2018-02-17-174656

## Related issues and discussion
#484 
 
 ## Screen-shots, if any
 
 ## Checklist
 
 Please make sure these boxes are checked before submitting your pull request - thanks!
 
 
 
 - [ x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 
 - [x ] If you have multiple commits please combine them into one commit by squashing them.
 
 - [x ] Read and understood the contribution guidelines .

### CODE REVIEW

File: app/src/main/res/layout/fragment_nutrition_product.xml [hunk 1]
```diff
@@ -53,21 +53,37 @@
...
+             android:layout_width="wrap_content"
+             android:visibility="gone"
+             android:layout_height="wrap_content"
+             android:layout_marginRight="@dimen/spacing_normal"
+             android:layout_marginLeft="@dimen/spacing_normal"
+             android:text="@string/txtNoData"
+             android:textIsSelectable="true"
+             android:textSize="@dimen/font_normal"
+             android:textStyle="bold" />
+     </LinearLayout>
```
Comment (jinqian) [inline app/src/main/res/layout/fragment_nutrition_product.xml]: I don't think it's a good idea to add a text view for no data case at fragment level. 

It's better to tweak the view binding process in `NutrientLevelListAdapter`. Considering the no data as one of the case and make your item view more resistant.

File: app/src/main/res/layout/fragment_nutrition_product.xml [hunk 2]
```diff
@@ -53,21 +53,37 @@
             android:padding="@dimen/spacing_small"
             android:textIsSelectable="true"
             android:textSize="@dimen/font_normal" />
+     <LinearLayout   android:layout_width="fill_parent"
```
[the remark this answers is not in the document]
Comment (sowmya9029) [reply app/src/main/res/layout/fragment_nutrition_product.xml]: removed this

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/NutritionProductFragment.java [hunk 3]
```diff
@@ -41,6 +41,7 @@
     @BindView(R.id.listNutrientLevels) ListView lv;
     @BindView(R.id.textServingSize) TextView serving;
     @BindView(R.id.textCarbonFootprint) TextView carbonFootprint;
+    @BindView(R.id.txtNoDataString)TextView noTextString;
```
[the remark this answers is not in the document]
Comment (sowmya9029) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/NutritionProductFragment.java]: fixed

File: app/src/main/res/layout/fragment_nutrition_product.xml (hunk 1 above)
Comment (sowmya9029) [reply app/src/main/res/layout/fragment_nutrition_product.xml]: I don't know how modifying the list adapter will help, If we want the "No data" text with the header.The other approach which I came up is to append "No data" text to the header in the NutritionProductFragment class.

**Comment by sowmya9029** [comment]
The comments have been addressed, please review this pull request @jinqian

**Comment by teolemon** [comment]
perhaps @ross-holloway94 or @policorn . @jinqian is having spotty Internet access those days.

**Comment by policorn** [comment]
@teolemon, this one can be merged

**Comment by teolemon** [comment]
Merged, thanks @sowmya9029 :-)

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/NutritionProductFragment.java [hunk 4]
```diff
@@ -74,7 +75,8 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         }
 
         if (fat == null && salt == null && saturatedFat == null && sugars == null) {
-            levelItem.add(new NutrientLevelItem(getString(R.string.txtNoData), "", "", R.drawable.error_image));
+            textNutrientTxt.append(" "+getString(R.string.txtNoData));
```
Comment (jinqian) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/NutritionProductFragment.java]: Well it's pity that this line is not correctly formatted, missing spaces around `+`

Comment (sowmya9029) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/NutritionProductFragment.java]: Thanks for pointing it out.I missed it.

**Pull request merged**